plugin/evm/limitorders/limit_orders_database.go

```diff
@@ -0,0 +1,157 @@
+	if err != nil {
+		fmt.Println(err)
+		return err
+	}
+	insertSQL := "INSERT INTO limit_orders(user_address, position_type, base_asset_quantity, price, salt, signature, status) VALUES (?, ?, ?, ?, ?, ?, 'open')"
+	statement, err := lod.db.Prepare(insertSQL)
+	if err != nil {
+		return err
+	}
+	_, err = statement.Exec(userAddress, positionType, baseAssetQuantity, price, salt, signature)
+	return err
+}
+
+func (lod *limitOrderDatabase) UpdateLimitOrderStatus(userAddress string, salt string, status string) error {
+	// TODO: validate inputs
+	updateSQL := "UPDATE limit_orders SET status = ? WHERE user_address = ? AND salt = ?"
+	statement, err := lod.db.Prepare(updateSQL)
+	if err != nil {
+		return err
+	}
+	_, err = statement.Exec(status, userAddress, salt)
+	return err
+}
+
+func (lod *limitOrderDatabase) GetLimitOrderByPositionTypeAndPrice(positionType string, price float64) []LimitOrder {
+	var rows = &sql.Rows{}
+	orderStatus := "open"
+	var limitOrders = []LimitOrder{}
+	if positionType == "short" {
+		rows = getShortLimitOrderByPrice(lod.db, price, orderStatus)
+	}
+	if positionType == "long" {
+		rows = getLongLimitOrderByPrice(lod.db, price, orderStatus)
+	}
+	defer rows.Close()
+	for rows.Next() {
+		var id uint64
+		var userAddress string
+		var baseAssetQuantity int
+		var price float64
+		var salt string
+		var signature []byte
+		_ = rows.Scan(&id, &userAddress, &baseAssetQuantity, &price, &salt, &signature)
+		limitOrder := &LimitOrder{
+			id:                id,
+			PositionType:      positionType,
+			UserAddress:       userAddress,
+			BaseAssetQuantity: baseAssetQuantity,
+			Price:             price,
+			Salt:              salt,
+			Signature:         signature,
+			Status:            orderStatus,
+		}
+		limitOrders = append(limitOrders, *limitOrder)
+	}
+	return limitOrders
+}
+
+func getShortLimitOrderByPrice(db *sql.DB, price float64, status string) *sql.Rows {
+	stmt, _ := db.Prepare(`SELECT id, user_address, base_asset_quantity, price, salt, signature
+		from limit_orders
+		where position_type = ? and price <= ? and status = ?`)
+	rows, _ := stmt.Query("short", price, status)
+	return rows
+}
+
+func getLongLimitOrderByPrice(db *sql.DB, price float64, status string) *sql.Rows {
+	stmt, _ := db.Prepare(`SELECT id, user_address, base_asset_quantity, price, salt, signature
+		from limit_orders
+		where position_type = ? and price >= ? and status = ?`)
+	rows, _ := stmt.Query("long", price, status)
+	return rows
+}
+
+func createTable(db *sql.DB) error {
+	createLimitOrderTableSql := `CREATE TABLE if not exists limit_orders (
+    	"id" integer NOT NULL PRIMARY KEY AUTOINCREMENT,
+		"position_type" VARCHAR(64) NOT NULL, 
+    	"user_address" VARCHAR(64) NOT NULL,
+    	"base_asset_quantity" INTEGER NOT NULL,
+    	"price" FLOAT NOT NULL,
+    	"status" VARCHAR(64) NOT NULL,
+    	"salt" VARCHAR(64) NOT NULL,
+		"signature" TEXT NOT NULL
+	);`
```

Comment:
Without a migration strat, this could get ugly during upgrades. Libs like GORM have this built-in, or https://github.com/golang-migrate/migrate https://github.com/pressly/goose